File: 9-palindrome-number/palindrome-number.c
bool isPalindrome(int x) {
    int copy = x;
    long reverse = 0;
    if(x < 0){
        return false;
    }
    while(copy!=0){
        int digit = copy%10;
        reverse = reverse * 10;
        reverse += digit;
        copy /= 10;
    }
    if(reverse == x){
        return true;
    }
    return false;
}

File: 9-palindrome-number/palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        reverse = 0
        copy = x
        if x < 0:
            return False
        while copy != 0:
            digit = copy % 10
            reverse = reverse * 10
            reverse += digit
            copy /= 10
        if reverse == x:
            return True
        return False
        